indexing-service/src/test/java/io/druid/indexing/overlord/http/OverlordResourceTest.java

```diff
@@ -0,0 +1,326 @@
+    response = overlordResource.getTaskStatus(taskId_0);
+    Assert.assertEquals(taskId_0, ((Map) response.getEntity()).get("task"));
+    Assert.assertEquals(
+        TaskStatus.running(taskId_0).getStatusCode(),
+        ((TaskStatus) ((Map) response.getEntity()).get("status")).getStatusCode()
+    );
+
+    // Simulate completion of task_0
+    taskCountDownLatches[Integer.parseInt(taskId_0)].countDown();
+    // Wait for taskQueue to handle success status of task_0
+    Thread.sleep(10);
+    response = overlordResource.getTaskStatus(taskId_0);
+    Assert.assertEquals(200, response.getStatus());
+    Assert.assertEquals(
+        TaskStatus.success(taskId_0).getStatusCode(),
+        ((TaskStatus) ((Map) response.getEntity()).get("status")).getStatusCode()
+    );
+
+    // Manually insert task in taskStorage
+    String taskId_1 = "1";
+    NoopTask task_1 = new NoopTask(taskId_1, 0, 0, null, null);
+    taskStorage.insert(task_1, TaskStatus.running(taskId_1));
+
+    response = overlordResource.getWaitingTasks();
+    // 1 task that was manually inserted would be in waiting state
+    Assert.assertEquals(1, (((List) response.getEntity()).size()));
+    // Wait for taskStorage sync
+    while (((List) (overlordResource.getRunningTasks().getEntity())).size() != 1) {
+      Thread.sleep(20);
+    }
+    // Simulate completion of task_1
+    taskCountDownLatches[Integer.parseInt(taskId_1)].countDown();
+    // Wait for taskQueue to handle success status of task_1
+    Thread.sleep(10);
+
+    // should return number of tasks which are not in running state
+    response = overlordResource.getCompleteTasks();
+    Assert.assertEquals(2, (((List) response.getEntity()).size()));
+    taskMaster.stop();
+    Assert.assertFalse(taskMaster.isLeading());
+    EasyMock.verify(taskLockbox, taskActionClientFactory);
+  }
+
+  @After
+  public void tearDown() throws Exception
+  {
+    tearDownServerAndCurator();
+  }
+
+  public static class MockTaskRunner implements TaskRunner
+  {
+    private CountDownLatch[] taskLatches;
+    private Map<Integer, TaskRunnerWorkItem> taskRunnerWorkItems;
+    private Map<Integer, TaskRunnerWorkItem> runningTaskRunnerWorkItems;
+
+    public MockTaskRunner(CountDownLatch[] taskLatches)
+    {
+      this.taskLatches = taskLatches;
+      this.taskRunnerWorkItems = new HashMap<>();
+      this.runningTaskRunnerWorkItems = new HashMap<>();
+    }
+
+    @Override
+    public ListenableFuture<TaskStatus> run(final Task task)
+    {
+      final int taskId = Integer.parseInt(task.getId());
+      ListenableFuture<TaskStatus> future = MoreExecutors.listeningDecorator(
+          Execs.singleThreaded(
+              "noop_test_task_exec_%s"
+          )
+      ).submit(
+          new Callable<TaskStatus>()
+          {
+            @Override
+            public TaskStatus call() throws Exception
+            {
+              try {
+                taskLatches[taskId].await(2, TimeUnit.SECONDS);
```

Comment:
yes...that's correct. I forgot to remove the timeout after adding the timeout for the test. I will correct it. Thanks